src/core/border_router/routing_manager.cpp

```diff
@@ -449,7 +449,11 @@ void RoutingManager::EvaluateOmrPrefix(OmrPrefixArray &aNewOmrPrefixes)
         // The `aNewOmrPrefixes` remains empty if we fail to publish
         // the local OMR prefix.
     }
-    else if (publishedLocalOmrPrefix != nullptr && smallestOmrPrefix != publishedLocalOmrPrefix)
+    else if (*smallestOmrPrefix == mLocalOmrPrefix)
+    {
+        IgnoreError(PublishLocalOmrPrefix());
+    }
+    else if (IsOmrPrefixAddedToLocalNetworkData())
```

Comment:
@abtink @wgtdkp 
The change here checks if `mLocalOmrPrefix` was added to the Local Network Data, instead of checking if `mLocalOmrPrefix` is added to the Leader's Network Data.
Here the BR has already decided that it should not be advertising `mLocalOmrPrefix`. Whether or not `mLocalOmrPrefix` is in Leader's Network Data is relevant. The BR only needs to remove `mLocalOmrPrefix` from Local Network Data and eventually Local Network Data will synchronize with Leader's Network Data. 

The original code also has a potential issue that the BR may send `NET_DATA.ntf` to Leader indefinitely when these conditions are met:
- `mLocalOmrPrefix` was added to Local Network Data
- `mLocalOmrPrefix` is not added to Leader' Network Data because there is no more space in Leader's Network Data
Since `publishedLocalOmrPrefix` will be `nullptr` in such case, the BR never try to remove `mLocalOmrPrefix` from Local Network Data.